FreeRTOS/Demo/CORTEX_MPS2_QEMU_IAR_GCC/.vscode/tasks.json

```diff
@@ -0,0 +1,41 @@
+{
+    // See https://go.microsoft.com/fwlink/?LinkId=733558
+    // for the documentation about the tasks.json format
+    "version": "2.0.0",
+    "tasks": [
+        {
+            "label": "Build QEMU",
+            "type": "shell",
+            "command": "sh -c -l 'make --directory=${workspaceFolder}/build/gcc' ",
+            "group": {
+                "kind": "build",
+                "isDefault": true
+            },
+            "problemMatcher": []
+        },
+        {
+            "label": "Run QEMU",
+            "type": "shell",
+            "command": "echo 'QEMU RTOSdemo started'; qemu-system-arm -machine mps2-an385 -cpu cortex-m3 -kernel ${workspaceFolder}/build/gcc/output/RTOSDemo.out -monitor none -nographic -serial stdio -s -S",
+            "dependsOn": ["Build QEMU"],
+            "isBackground": true,
+            "problemMatcher": [
+                {
+                  "pattern": [
+                    {
+                      "regexp": "QEMU RTOSdemo started",
```

Comment:
VSCode needs a something from a background task to know it is "completed" in order to continue and run the launch config which attaches the debugger. "qemu-system-arm" has no way of producing output to say it is ready, and if the debugger is to be attached before the demo begins, adding output in the demo code cannot be relied upon.

This is a way of getting artificial output, assuming the launch succeeds. A couple of alternative methods could be:

1. Creating a new task to be the prelaunch task. The new task would depend on this task and run a bash script to detect if the qemu process was created. This ensures if the debugger attempts to attach it will always find a successfully launched program. However, an echo message in the "Run QEMU" task will still be necessary to signal VSCode to run this new task, and the new task would need a timeout implementation or risk checking for the qemu process before the "Run QEMU" task has executed the command (gdb target remote already has a timeout so this is not an issue in the current method).

2. Have the user manually start the run task, and then run the launch config. The code part of this might be more intuitive to understand, but the user would have to perform twice as many clicks every time they wanted to run the program.

Reply:
Alright. Thank you for the detailed explanation. Let's say for some reason "qemu-system-arm" fails, would the user get a message to be able to find that?

Reply:
Under the current implementation, if the command "qemu-system-arm" fails the error message will be printed in the integrated terminal, which is visible by default when the task is getting run.

Reply:
Perfect! Thanks.